test/e2e/operator_test.go

```diff
@@ -121,6 +125,9 @@ func TestMain(m *testing.M) {
 		os.Exit(1)
 	}
 
+	if version.SHORTCOMMIT != "" {
+		hostedZoneDomain = version.SHORTCOMMIT + "." + baseZoneDomain
+	}
```

Comment:
Why this change? `SHORTCOMMIT` is supposed to be there, it's set at compile time [here](https://github.com/openshift/external-dns-operator/blob/main/Makefile#L87).

Reply:
yes, since it is compile time.
when we execute in local, this value will be empty so the `hostedZoneDomain` look like ".example-test.info"
which is invalid for azure to create zone.

Reply:
After the discussion: the symbol may be not there when the test is run from IDE or for debugging. Let's do this check the, thanks @Connect2naga!